kpatch-build/lookup.c

```diff
@@ -108,9 +107,15 @@ static int locals_match(struct lookup_table *table, int idx,
 			continue;
 
 		found = 0;
-		for (child = child_locals; child->name; child++) {
-			if (child->type == sym->type &&
-			    !strcmp(child->name, sym->name)) {
+		iter_sym = file_sym;
+		list_for_each_entry_continue(iter_sym, sym_list, list) {
+			if (iter_sym->type == STT_FILE)
+				break;
+			if (iter_sym->bind != STB_LOCAL)
+				continue;
+
```

Comment:
To match the previous functionality I think we also want to make sure it's either an STT_FUNC or an STT_OBJECT (and otherwise `continue`).

Reply:
This is actually intentional, so is the removal in the next comment. The very next statement is `iter_sym->type == sym->type` which does make sure the type is one of those possibilities so the check was redundant. Or do you think it is worth keeping it for clarity?

Reply:
Ah I see now.  Never mind...